### ISSUE TITLE
jekyll 3.0.3 rendered post files error when collections: null

### ISSUE THREAD

**Issue by Treri** [jekyll/jekyll#4550]
**Latest edit**:
Found reason. `collections: null` will lead github used jekyll (version 3.0.3) generate no rendered posts files

###### What version of Jekyll are you using (`jekyll -v`)?
3.0.3

###### What operating system are you using?
OS X 10.11.2

###### What did you do?
(Please include the content causing the issue, any relevant configuration settings, and the command you ran)
just run `jekyll serve` or `jekyll build`

then the build directory structure is below
```
.
├── CNAME
├── README.md
├── _config.yml
├── _includes
│   ├── comments
│   └── comments-providers
├── _layouts
│   ├── default.html
│   └── post.html
├── _posts
│   └── 2014-01-04-iLotus.md
├── _site
│   ├── CNAME
│   ├── README.md
│   ├── archives.html
│   ├── categories.html
│   ├── contact.html
│   ├── feed.xml
│   ├── index.html
│   ├── links.html
│   ├── sitemap.txt
│   ├── sitemap.xml
│   ├── static
│   └── tags.html
├── archives.html
├── categories.html
├── contact.html
├── feed.xml
├── index.html
├── links.html
├── sitemap.txt
├── sitemap.xml
├── static
│   ├── font
│   ├── iLotus2.js
│   ├── images
│   ├── js
│   ├── prettify.js
│   └── style.css
└── tags.html
```

my _config.yml is below 

```yml
# Where things are
source: .
destination: ./_site
plugins_dir: ./_plugins
layouts_dir: ./_layouts
data_dir: ./_data
includes_dir: ./_includes
collections: null

# Handling Reading
safe: false
include: [".htaccess"]
exclude: []
keep_files: [".git", ".svn"]
encoding: "utf-8"
markdown_ext: "markdown,mkdown,mkdn,mkd,md"

# Filtering Content
show_drafts: null
limit_posts: 0
future: true
unpublished: false

# Plugins
whitelist: []
gems: []

# Conversion
markdown: kramdown
highlighter: rouge
lsi: false
excerpt_separator: "\n\n"
incremental: false

# Serving
detach: false
port: 4000
host: 127.0.0.1
baseurl: "" # does not include hostname

# Outputting
permalink: /:year/:month/:title.html
# paginate: 8
paginate_path: /page:num
timezone: Asia/Shanghai

quiet: false
defaults: []

# Markdown Processors
kramdown:
  auto_ids: true
  footnote_nr: 1
  entity_output: as_char
  toc_levels: 1..6
  smart_quotes: lsquo,rsquo,ldquo,rdquo
  enable_coderay: false

  coderay:
    coderay_wrap: div
    coderay_line_numbers: inline
    coderay_line_number_start: 1
    coderay_tab_width: 4
    coderay_bold_every: 10
    coderay_css: style

#iLotus configs
iLotus:
  columns: 1
  begin: 2011

  # comments provider
  comments:
    provider: duoshuo
    duoshuo:
      short_name: short_name

#站点基础配置
title: iLotus
subTitle: Just another Jekyll blog.
description: 这是 PIZn 的又一个博客，它是基于 Jekyll 搭建起来的。在这里，不仅仅是用来快速记录工作，学习，生活的一点一滴的地方，更是 PIZn 对 Jekyll 的学习和使用的一次有意义的实践。
keywords: PIZn, GitHub, Jekyll, Plugins, Works, Archives, Quotes, Violet, css, html, javascript, wordpress, logo, design, geek
url: http://localhost:4000
feed: /feed.xml

#菜单配置
nav:
  - text: 首页
    url: /index.html
    icon: icon-home
    class: home
  - text: 文章归档
    url: /archives.html
    icon: icon-reorder
    class: none
  - text: 关于我
    url: /contact.html
    icon: icon-envelope-alt
    class: none
  - text: 联系我
    url: http://www.zhanxin.info/contact.html
    icon: icon-heart
    class: none
  - text: 博客源代码
    url: https://github.com/pizn/iLotus
    icon: icon-github
    class: none

author:
  name: PIZn

```

###### What did you expect to see?
correct rendered post files wrote to disk located at `_site/:year/:month/:title.html` and index.html have right post links

###### What did you see instead?
all other files are processed and wrote to `_site` directory, except these posts files.
In other words, no any post files wrote to disk. But rendered index.html have post list with wrong post link

PS. My repo is https://github.com/Treri/iLotus-jekyll 

You can just clone it locally and use jekyll version 3.0.3 to reproduce the error

I have the same error on github pages blog.

**Comment by Crunch09** [comment]
If i didn't misuse `git bisect`, 5d79c55b2cee0aa19ca57ad331333edba9db47b7 fixed this behavior

**Comment by Treri** [comment]
when will github update its jekyll version? if the used version is still 3.0.3, I have to comment out that config field to make github pages work as expected.
